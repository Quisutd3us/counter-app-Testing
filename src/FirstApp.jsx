import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ({title, name, subTitle,}) => {
  return (
      <div>
        <h1 data-testid={'test-title'}> { title } </h1>
        <h3 data-testid={'test-subtitle'}>{subTitle}</h3>
        <p>{name}</p>
      </div>
  );
};

FirstApp.propTypes = {
  title:PropTypes.string.isRequired,
  name:PropTypes.string.isRequired,
  subTitle:PropTypes.string.isRequired
};

FirstApp.defaultProps={
  name:'Dont have name',
  subTitle:'Dont have subtitle',
  //title:'Dont have title',
}

export default FirstApp;