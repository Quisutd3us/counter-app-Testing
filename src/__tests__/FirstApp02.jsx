import {render, screen} from "@testing-library/react";
import FirstApp from "../FirstApp.jsx";

describe('Test for component <FirstApp/>', () => {
  const title = 'This is the Title ..'
  const subTitle = 'This is the SubTitle ..'
  test('Must be Match with the Snapshot', () => {
    const {container} = render(
        <FirstApp
            title={'title'}
        />
    )
    expect(container).toMatchSnapshot()
  })

  test('Must be the message "This is the Title .."', () => {
    render(
        <FirstApp
            title={title}
        />
    )
    expect(screen.getByText(title)).toBeTruthy()
  })
  test('Must be show the title on H1 tag', () => {
    render(
        <FirstApp
            title={title}
        />
    )
    expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title)
  })

  test('Must be show the subtitle send by props',()=>{
    render(
        <FirstApp
            title={title}
            subTitle={subTitle}
        />
    )
    expect(screen.getByText(subTitle).innerHTML).toContain(subTitle)
  })
})