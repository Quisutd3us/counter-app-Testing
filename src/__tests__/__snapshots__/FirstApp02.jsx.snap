// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for component <FirstApp/> Must be Match with the Snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
       
      title
       
    </h1>
    <h3
      data-testid="test-subtitle"
    >
      Dont have subtitle
    </h3>
    <p>
      Dont have name
    </p>
  </div>
</div>
`;
